application/ui/src/features/inspect/dataset/dataset-item/dataset-item.component.tsx

```diff
@@ -1,38 +1,45 @@
-import { Image } from '@geti-inspect/icons';
-import { Flex } from '@geti/ui';
+import { useEffect } from 'react';
+
+import { Skeleton } from '@geti/ui';
+import { useQuery } from '@tanstack/react-query';
 import { clsx } from 'clsx';
 
 import { useInference } from '../../inference-provider.component';
 import { useSelectedMediaItem } from '../../selected-media-item-provider.component';
+import { DeleteMediaItem } from '../delete-dataset-item/delete-dataset-item.component';
 import { type MediaItem } from '../types';
 
 import styles from './dataset-item.module.scss';
 
-const DatasetItemPlaceholder = () => {
-    return (
-        <Flex
-            justifyContent={'center'}
-            alignItems={'center'}
-            UNSAFE_className={clsx(styles.datasetItemPlaceholder, styles.datasetItem)}
-        >
-            <Flex>
-                <Image />
-            </Flex>
-        </Flex>
-    );
-};
-
 interface DatasetItemProps {
     mediaItem: MediaItem;
 }
 
-const DatasetItem = ({ mediaItem }: DatasetItemProps) => {
+export const DatasetItem = ({ mediaItem }: DatasetItemProps) => {
     const { selectedMediaItem, onSetSelectedMediaItem } = useSelectedMediaItem();
     const { onInference, selectedModelId } = useInference();
 
     const isSelected = selectedMediaItem?.id === mediaItem.id;
 
-    const mediaUrl = `/api/projects/${mediaItem.project_id}/images/${mediaItem.id}/thumbnail`;
+    const { data: thumbnailBlob, isLoading } = useQuery({
+        queryKey: ['media', mediaItem.id],
+        queryFn: async () => {
+            const response = await fetch(`/api/projects/${mediaItem.project_id}/images/${mediaItem.id}/thumbnail`);
```

Comment:
Why not use `$api.useQuery` or `fetchClient.fetch()` here?

Reply:
`$api` is configured to expect JSON responses, which always results in an error in this case. I looked into it and it would require a custom fetch with some conditions. After trying a few approaches and not getting it to work, I went with the simpler solution.